indexing-service/src/main/java/io/druid/indexing/common/TaskLock.java

```diff
@@ -31,19 +31,25 @@
   private final String dataSource;
   private final Interval interval;
   private final String version;
+  private final Integer priority;
+  private final Boolean preemptive;
 
   @JsonCreator
   public TaskLock(
       @JsonProperty("groupId") String groupId,
       @JsonProperty("dataSource") String dataSource,
       @JsonProperty("interval") Interval interval,
-      @JsonProperty("version") String version
+      @JsonProperty("version") String version,
+      @JsonProperty("priority") Integer priority,
+      @JsonProperty("preemptive") Boolean preemptive
```

Comment:
Actually when I first wrote the code I use a variable named `exclusiveLock` instead of `preemptive` and the default value was "false" and after upgrading it becomes "true" so I am OK with negating the logic in regards to this variable. However, IIRC someone asked specifically to change this to `preemptive` but I do not recall who.